### PR TITLE
Support converting blas lstm to dnn lstm

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?

Support converting blas lstm to dnn lstm

## How was this patch tested?

unit tests

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BiRecurrent.scala [hunk 1]
```diff
@@ -62,8 +62,11 @@ class BiRecurrent[T : ClassTag] (
         .add(Reverse[T](timeDim))
         .add(revLayer)
         .add(Reverse[T](timeDim))))
-  if (merge == null) birnn.add(CAddTable[T](true))
-  else birnn.add(merge)
+  if (merge == null) merge = CAddTable[T](true)
+  birnn.add(merge)
+
+
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BiRecurrent.scala]: remove empty line?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala [hunk 2]
```diff
@@ -293,6 +293,12 @@ class RNN(
       updateOutputTensors = buffer.toArray
     }
 
+    if (isTraining()) {
+      weight.sync()
+      weight_i.sync()
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala]: what does `weight_i` mean ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BiRecurrent.scala (hunk 1 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BiRecurrent.scala]: ok

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala (hunk 2 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala]: dnn lstm has two weights, one is weight, and another is iteration weight.

**Comment by zhangxiaoli73** [comment]
jenkins pass: http://mlt-bddl01.sh.intel.com:8080/view/PR/job/BigDL-PRVN-lite/702/

**Pull request merged**